tests/functional/adapter/test_optional_iceberg.py

```diff
@@ -0,0 +1,232 @@
+"""
```

Comment:
shall we call this test_iceberg_incremental.py??

Reply:
For what I seen in the tests - other than checking unique location, the other test cases are based mostly on incremental merge, so test_incremental_iceberg.py is good.